File: Python3/ValidParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        Map = {")": "(", "]": "[", "}": "{"}
        stack = []

        for c in s:
            if c not in Map:
                stack.append(c)
                continue
            if not stack or stack[-1] != Map[c]:
                return False
            stack.pop()

        return not stack


#2

class Solution:
    def isValid(self, s: str) -> bool:
        Map = {"(": ")", "[": "]", "{": "}"}
        stack = []

        for c in s:
            if c in Map:
                stack.append(c)
            elif len(stack) > 0 and Map[stack[-1]] == c :
                stack.pop()
            else:
                return False

        return len(stack)==0

#3

class Solution:
    def isValid(self, s: str) -> bool:
        mapping = {")":"(", "]":"[", "}":"{"}
        stack=[]

        for c in s:
            if c in mapping and len(stack) > 0:
                v = stack.pop()
                if mapping[c] != v:
                    return False
            else:
                stack.append(c)
        
        return len(stack) == 0

File: Golang/ValidParentheses.go
func isValid(s string) bool {
    m := make(map[rune]rune)

    m['('] = ')'
    m['['] = ']'
    m['{'] = '}'
    
    stack := make([]rune,0)
    for _, v := range s {
        if _, ok := m[v]; ok {
            stack = append(stack, v)
        } else if len(stack) > 0 && m[stack[len(stack)-1]] == v {
            stack = stack[:len(stack)-1]
        } else {
            return false
        }
    }

    return len(stack) == 0

}

#2

func isValid(s string) bool {
    mapping := make(map[rune]rune)
    mapping[')'] = '('
    mapping[']'] = '['
    mapping['}'] = '{'
    stack := make([]rune,0)

    for _, c := range s {
        if _, ok := mapping[c]; ok && len(stack) > 0 {
            if mapping[c] == stack[len(stack)-1] {
                stack = stack[:len(stack)-1]
                continue
            } else {
                stack = stack[:len(stack)-1]
                return false
            }
        } else {
            stack = append(stack, c)
        }
    }
    return len(stack) == 0
}